slice.go

```diff
@@ -693,3 +694,77 @@ func Splice[T any, Slice ~[]T](collection Slice, i int, elements ...T) Slice {
 
 	return append(append(append(output, collection[:i]...), elements...), collection[i:]...)
 }
+
+// Pull removes all given values from slice using SameValueZero for equality comparisons.
+func Pull[T comparable, Slice ~[]T](collection Slice, elements ...T) Slice {
+	output := make(Slice, 0, len(collection))
+
+	for _, item := range collection {
+		if !slices.Contains(elements, item) {
+			output = append(output, item)
+		}
+	}
+
+	return output
+}
+
+// PullAll is like [Pull] except that it accepts an array of values to remove
+func PullAll[T comparable, Slice ~[]T](collection Slice, elements ...T) Slice {
+	output := make(Slice, 0, len(collection))
+
+	for _, item := range collection {
+		if !slices.Contains(elements, item) {
+			output = append(output, item)
+		}
+	}
+
+	return output
+}
+
+// PullAllBy is like [PullAll] except that it accepts iteratee which is invoked
+// for each element of slice and values to generate the criterion by which they're compared
+func PullAllBy[T any, U comparable, Slice ~[]T](collection Slice, elements []U, iteratee func(T) U) Slice {
+	elementCriteria := make(map[U]struct{}, len(elements))
+	for _, element := range elements {
+		elementCriteria[element] = struct{}{}
+	}
+
+	var output Slice
+	for _, item := range collection {
+		if _, shouldRemove := elementCriteria[iteratee(item)]; !shouldRemove {
+			output = append(output, item)
+		}
+	}
+
+	return output
+}
+
+// PullAt removes elements from slice corresponding to indexes and returns
+// an array of removed elements
+func PullAt[T any](slice []T, indexes []int) ([]T, []T) {
```

Comment:
A variadic could be used here too

```suggestion
func PullAt[T any](slice []T, indexes ..
int) ([]T, []T) {
```

Reply:
Done